### PR TITLE
[ObjCRuntime] Remove XM limitation. Fixes #55857.

### PR DESCRIPTION
**Pull request by rolfbjarne**
I can't remember why I added this limitation, git history provides no clues,
and it compiles fine.

Also run the xammac tests both in Debug and Release configurations (previously
only Debug; the Release configuration triggers this bug).

https://bugzilla.xamarin.com/show_bug.cgi?id=55857

### CODE REVIEW

File: src/ObjCRuntime/Runtime.cs [hunk 1]
```diff
@@ -575,9 +575,6 @@ static unsafe IntPtr GetMethodFromToken (uint token_ref)
 
 		static unsafe IntPtr GetGenericMethodFromToken (IntPtr obj, uint token_ref)
 		{
-#if MONOMAC
-			throw new NotSupportedException ();
```
Comment (spouliot) [inline src/ObjCRuntime/Runtime.cs]: maybe it was because XM classic does not support generics ?

Comment (spouliot) [inline src/ObjCRuntime/Runtime.cs]: if that's the case then maybe `#if MONOMAC && !XAMCORE_2_0` would be more appropriate

Comment (chamons) [inline src/ObjCRuntime/Runtime.cs]: I could imagine that classic was the reason for this.

Comment (rolfbjarne) [inline src/ObjCRuntime/Runtime.cs]: XI/Classic didn't support generics either. If generics was the issue, the ifdef would have been `#if !XAMCORE_2_0`, not `#if MONOMAC`.

Also I can't see how removing an unconditional exception can break things... and the static registrar will show errors at compile time for generics in Classic mode.

My guess is that it was a temporary ifdef I added while fixing something to make the code compile, and then the ifdef was never removed.

Comment (chamons) [inline src/ObjCRuntime/Runtime.cs]: I accept that logic. 👍

**Comment by rolfbjarne** [comment]
build

**Comment by rolfbjarne** [comment]
Test failures are unrelated:

* link all/watchOS: bug #[55719](https://bugzilla.xamarin.com/show_bug.cgi?id=55719)
* monotouch-test + MSBuild/iOS: signing failures to due bot misconfiguration (RE [notified](https://trello.com/c/gZXaChZt/989-no-ios-certificates-on-xam-mac-mini-1))

**Pull request merged**